cmd/system-probe/config/config.go

```diff
@@ -99,6 +99,15 @@ func newSysprobeConfig(configPath string, loadSecrets bool) (*Config, error) {
 	// load the configuration
 	_, err := aconfig.LoadCustom(aconfig.SystemProbe, "system-probe", loadSecrets, aconfig.Datadog.GetEnvVars())
 	if err != nil {
+		// System probe is not supported on darwin, so we should fail gracefully in this case.
+		if runtime.GOOS != "darwin" {
+			if errors.Is(err, os.ErrPermission) {
+				log.Warnf("Error loading config: %v (check config file permissions for dd-agent user)", err)
+			} else {
+				log.Warnf("Error loading config: %v", err)
+			}
+		}
```

Comment:
Why not just return a specific error stating that it's not supported on Darwin at the beginning of the error handling ?

It would allow not to care about Darwin after that, and also avoid returning an `unable to load system-probe config file` error when actually it's just not supported.

Reply:
Hi @pgimalac, Corrina followed Bryce's suggestion here to not log any message at all if we're trying to open `system-probe.yaml` on macOS(darwin) since it's not supported on this platform:
https://github.com/DataDog/datadog-agent/pull/16745#pullrequestreview-1398477034

For all remaining platforms, we're still logging the warn messages if there's a problem opening the file.

Reply:
Talked offline: the gist is that on Darwin we will have a `os.ErrNotExist` so the function will return cleanly (we actually don't return an error)
Thanks !